core/src/main/java/hudson/tasks/Maven.java

```diff
@@ -137,6 +137,12 @@
      */
     private GlobalSettingsProvider globalSettings;
 
+    /**
+     * Inject build variables as properties into maven process.
```

Comment:
I think you need add `ALL build vars` as it the main disaster. And note that escaped vars may cause issues.